samples/cpp/hello_world/src/main.cpp

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iostream>
+
+int main(void)
+{
+	std::cout << "Hello C++, world! " << CONFIG_BOARD << std::endl;
```

Comment:
Does this get routed to printk?

Reply:
Not 100% sure, but I feel like it should Just Work™. @stephanosio - any idea?

Reply:
> AFAIK std::endl includes a flush.
> That's why you should not use it to much 😅
> 
> Isn't iostream bloating? The last time I tried iostream added some 80k because of locale.

Yea, exactly. That's why I originally avoided it in the first rev of this PR. C++ hello world could just `#include` C hello world. I further expanded the original hello world so that you could add arbitrary languages with a couple lines of code, which was (IMHO) kind of short and sweet and to the point.

Then someone said it should be in a different project.

Then someone requested iostream, and so I did both iostream and printf versions.

Then someone said it introduced too much test complexity.

Then someone said it introduced too much build complexity.

In the end, I thought about just leaving a comment, and then realized that my editorial isn't even really necessary. Users can just figure out iostream is bloaty from existing sources and / or experimentation.

Then again, maybe someday someone will make a small-ish implementation of iostream (probably already exists - e.g. pigweed).

Future / not my problem.

Reply:
> > AFAIK std::endl includes a flush.
> > That's why you should not use it to much 😅
> > Isn't iostream bloating? The last time I tried iostream added some 80k because of locale.
> 
> Yea, exactly. That's why I originally avoided it in the first rev of this PR. C++ hello world could just `#include` C hello world. I further expanded the original hello world so that you could add arbitrary languages with a couple lines of code, which was (IMHO) kind of short and sweet and to the point.
> 
> Then someone said it should be in a different project.
> 
> Then someone requested iostream, and so I did both iostream and printf versions.
> 
> Then someone said it introduced too much test complexity.
> 
> Then someone said it introduced too much build complexity.
> 
> In the end, I thought about just leaving a comment, and then realized that my editorial isn't even really necessary. Users can just figure out iostream is bloaty from existing sources and / or experimentation.
> 
> Then again, maybe someday someone will make a small-ish implementation of iostream (probably already exists - e.g. pigweed).
> 
> Future / not my problem.

Hey, this just shows a working review process, even for something as simple as a hello world.

The iostream thing you could have dismissed as feature bloat and continued with a simple implementation, I did not see this being a firm request.

Reply:
> The iostream thing you could have dismissed as feature bloat and continued with a simple implementation, I did not see this being a firm request.

https://github.com/zephyrproject-rtos/zephyr/pull/65769#discussion_r1405249361

@nashif -

Reply:
A regex comma was misplaced 😅